src/core/utils/hooks/useIntersection.tsx

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+/**
+ * Hook that uses `IntersectionObserver` to trigger `onIntersect` callback when element is in view.
+ * Callback is called only once.
+ */
+export const useIntersection = (
+  onIntersect: () => void,
+  options: IntersectionObserverInit = {},
+) => {
+  const { root, rootMargin, threshold } = options;
```

Comment:
tests?

Reply:
Added for `rootMargin`, too lazy to add for others because we are not using them in Table 😀

Reply:
I meant to have tests file for only useIntersection hook.. in case we reuse it somewhere else.. and in table leave the tests which only check that needed callbacks are called...

Reply:
Test would be identical because I still need to mock `IntersectionObserver`.